pkg/file/pipeline/stamp/stamp.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stamp
+
+import (
+	"github.com/ethersphere/bee/pkg/file/pipeline"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type stampWriter struct {
+	stamper postage.Stamper
+	next    pipeline.ChainWriter
+}
+
+func NewStampWriter(stamper postage.Stamper, next pipeline.ChainWriter) pipeline.ChainWriter {
+	return &stampWriter{stamper: stamper, next: next}
+}
+
+func (w *stampWriter) ChainWrite(p *pipeline.PipeWriteArgs) (err error) {
+	addr := swarm.NewAddress(p.Ref)
+	p.Stamp, err = w.stamper.Stamp(addr)
```

Comment:
this sets the stamp on the pipe write args, but does not implement the change which adds the stamp to the actual chunk which is in turn stored on the localstore. i suggest that this change is done in one PR (change to the chunk interface + persistence in localstore)